src/gbxml/ReverseTranslator.cpp

```diff
@@ -1046,6 +1020,30 @@ namespace gbxml {
     return result;
   }
 
+  boost::optional<openstudio::model::ModelObject> ReverseTranslator::translateId(const pugi::xml_node& element,
+                                                                                 openstudio::model::ModelObject& modelObject) {
+    std::string id = element.attribute("id").value();
+    m_idToObjectMap.insert(std::make_pair(id, modelObject));
+    modelObject.setName(id);
+
+    model::AdditionalProperties result = modelObject.additionalProperties();
+    result.setFeature("gbXMLId", id);
+
+    return result;
+  }
```

Comment:
I know you're matching existing code, but the return type bothers me a bit here as well.
It's always initialized to an `AdditionalProperties` object. And it will never be used.